ledger/block/src/verify.rs

```diff
@@ -410,6 +410,13 @@ impl<N: Network> Block<N> {
             bail!("Found a duplicate transition in block {height}");
         }
 
+        // Ensure there are not duplicate program IDs.
+        if has_duplicates(
+            self.transactions().iter().filter_map(|tx| tx.transaction().deployment().map(|d| d.program_id())),
```

Comment:
This call checks ALL accepted and rejected deployments are unique. I think this is possible to fail. We should only be checking the accepted no?

Reply:
I believe this check only checks the accepted deployments. 

`ConfirmedTransaction::transaction()` gives you the underlying transaction, which is a `Deployment`, `Execution`, or `Fee`. Then the `Transaction::deployment` method returns a Some(Deployment) only for the deployment type, which excludes the rejected deployments.